projects/ng-components/lib/toggle-selector/toggle-selector.component.spec.ts

```diff
@@ -0,0 +1,172 @@
+        component.error = true;
+        fixture.detectChanges();
+        const container: DebugElement = fixture.debugElement.query(By.css(".error"));
+        const span: DebugElement = fixture.debugElement.query(By.css(".text-danger"));
+        expect(container).toBeDefined();
+        expect(container).not.toBeNull();
+        expect(span).toBeDefined();
+        expect(span).not.toBeNull();
+    });
+
+    describe("Things that should happen when multi is true", () => {
+        beforeEach(() => {
+            component.multi = true;
+            component.list = [
+                { value: "1", label: "Four mississipi" },
+                { value: "2", label: "Five mississipi" },
+            ];
+            fixture.detectChanges();
+        });
+
+        it("Should have type checkbox when multi is true", () => {
+            const elements: DebugElement[] = fixture.debugElement.queryAll(By.css(".toggle-selector input[type=checkbox]"));
+            expect(elements.length).toBe(component.list.length);
+        });
+
+        it("Should have two checked when both options are clicked", () => {
+            const element: HTMLElement[] = fixture.nativeElement.querySelectorAll(".toggle-selector");
+            element[1].dispatchEvent(new Event("click"));
+            fixture.detectChanges();
+            const inputEl: DebugElement[] = fixture.debugElement.queryAll(By.css("input"));
+            expect(inputEl.every(el => el.nativeElement.checked === true)).toBeFalsy();
+
+            element[0].dispatchEvent(new Event("click"));
+            fixture.detectChanges();
+            expect(inputEl.every(el => el.nativeElement.checked === true)).toBeTruthy();
+        });
+
+        it("should uncheck when previously checked option is clicked", () => {
+            const element: HTMLElement[] = fixture.nativeElement.querySelectorAll(".toggle-selector");
+            element[1].dispatchEvent(new Event("click"));
+            fixture.detectChanges();
+            const inputEl: DebugElement[] = fixture.debugElement.queryAll(By.css("input"));
+
+            expect(inputEl.every(el => el.nativeElement.checked === true)).toBeFalsy();
+
+            element[1].dispatchEvent(new Event("click"));
+            fixture.detectChanges();
+            expect(inputEl.some(el => el.nativeElement.checked === true)).toBeFalsy();
+        });
+    });
+
+    describe("Things that should happen when multi is false", () => {
+        beforeEach(() => {
+            component.multi = false;
+            component.list = [
+                { value: "1", label: "Four mississipi" },
+                { value: "2", label: "Five mississipi" },
+            ];
+            fixture.detectChanges();
+        });
+
+        it("Should have type radio when multi is false", () => {
+            const elements: DebugElement[] = fixture.debugElement.queryAll(By.css(".toggle-selector input[type=radio]"));
+            expect(elements.length).toBe(component.list.length);
```

Comment:
I think this is cleaner and easier to read
```suggestion
            expect(elements).toHaveLength(component.list.length);
```

Reply:
This throws an error Property 'toHaveLength' does not exist on type 'ArrayLikeMatchers<DebugElement>'